File: WzorceProjektowe.API/Controllers/PatternController.cs
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using WzorceProjektowe.API.Dto;
using WzorceProjektowe.API.Interfaces;

namespace WzorceProjektowe.API.Controllers
{
    [Route("api/[controller]")]
    [ApiController]
    public class PatternController : ControllerBase
    {
        private readonly IPatternService _patternService;

        public PatternController(IPatternService patternService)
        {
            _patternService = patternService;
        }

        [HttpPost("GetPatternsByType")]
        public async Task<IActionResult> GetPatternsByType([FromBody]GetPatternByTypeRequestDto type)
        {
            return await _patternService.GetPatternsByType(type.Type);
        }

        [HttpGet("{id}")]
        public async Task<IActionResult> GetPatternById(Guid guid)
        {
            return await _patternService.GetPatternById(guid);
        }

        [HttpPost("GetPatternCodeByType")]
        public async Task<IActionResult> GetPatternCodeByType([FromBody] GetPatternCodeByTypeRequestDto request)
        {
            return await _patternService.GetPatternCodeByType(request);
        }
        [HttpPost("GetPatternCodeByName")]
        public async Task<IActionResult> GetPatternCodeByName([FromBody] GetPatternCodeByNameRequestDto request)
        {
            return await _patternService.GetPatternCodeByName(request);
        }
        [HttpPost("DownloadCode")]
        public async Task<IActionResult> DownloadCode([FromBody] DownloadCodeRequestDto request)
        {
            try
            {
                byte[] zipBytes = await _patternService.DownloadCode(request);

                // Return the ZIP file as response
                return File(zipBytes, "application/zip", $"{request.PatternName}.zip");
            }
            catch (Exception ex)
            {
                // Handle any exceptions and return appropriate response
                return StatusCode(500, $"An error occurred: {ex.Message}");
            }
        }

    }
}

File: wzorceprojektowe.fe/src/Controllers/PatternController.js
import PatternService from "../Services/PatternService";

export const GetPatternById = async (data) => {
  const response = await PatternService.GetPatternById(data);
  if (response === undefined) {
    return null;
  }
  if (response.status === 200) {
    return response.data;
  } else if (response.status === 400) {
    return null;
  }
};
export const GetPatternsByType = async (data) => {
    const response = await PatternService.GetPatternsByType(data);
    if (response === undefined) {
      return null;
    }
    if (response.status === 200) {
      return response.data;
    } else if (response.status === 400) {
      return null;
    }
  };
  export const GetPatternCodeById = async (data) => {
    const response = await PatternService.GetPatternCodeById(data);
    if (response === undefined) {
      return null;
    }
    if (response.status === 200) {
      return response.data;
    } else if (response.status === 400) {
      return null;
    }
  };
  export const GetPatternCodeByName = async (data) => {
    const response = await PatternService.GetPatternCodeByName(data);
    if (response === undefined) {
      return null;
    }
    if (response.status === 200) {
      return response.data;
    } else if (response.status === 400 || response.status === 404) {
      alert("To be continued...")
      return null;
    }
  };
  export const DownloadCode = async (data) => {
    const response = await PatternService.DownloadCode(data);
    if (response === undefined) {
      return null;
    }
    if (response.status === 200) {
      return response.data;
    } else if (response.status === 400) {
      alert("Error occured while downloading file.");
      return null;
    }
  };